packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -115,6 +115,119 @@ describe('<DataGrid /> - Filter', () => {
       });
       expect(getColumnValues()).to.deep.equal(['Nike']);
     });
+
+    [
+      { operator: 'contains', value: 'a', expected: ['Asics'] },
+      { operator: 'startsWith', value: 'r', expected: ['RedBull'] },
+      { operator: 'equals', value: 'hugo', expected: ['Hugo'] },
+      { operator: 'endsWith', value: 'ics', expected: ['Asics'] },
+    ].forEach(({ operator, value, expected }) => {
+      it(`should allow object as value and work with valueGetter, operator: ${operator}`, () => {
+        render(
+          <TestCase
+            value={value}
+            operator={operator}
+            rows={[
+              {
+                id: 3,
+                brand: { name: 'Asics' },
+              },
+              {
+                id: 4,
+                brand: { name: 'RedBull' },
+              },
+              {
+                id: 5,
+                brand: { name: 'Hugo' },
+              },
+            ]}
+            columns={[{ field: 'brand', valueGetter: (params) => params.value.name }]}
+          />,
+        );
+        expect(getColumnValues()).to.deep.equal(expected);
+      });
+    });
+  });
+
+  describe('Numeric operators', () => {
+    [
+      { operator: '=', value: 1984, expected: [1984] },
+      { operator: '!=', value: 1984, expected: [1954, 1974] },
+      { operator: '>', value: 1974, expected: [1984] },
+      { operator: '>=', value: 1974, expected: [1984, 1974] },
+      { operator: '<', value: 1974, expected: [1954] },
+      { operator: '<=', value: 1974, expected: [1954, 1974] },
+    ].forEach(({ operator, value, expected }) => {
+      it(`should allow object as value and work with valueGetter, operator: ${operator}`, () => {
+        render(
+          <TestCase
+            value={value.toString()}
+            operator={operator}
+            rows={[
+              {
+                id: 3,
+                brand: { year: 1984 },
+              },
+              {
+                id: 4,
+                brand: { year: 1954 },
+              },
+              {
+                id: 5,
+                brand: { year: 1974 },
+              },
+            ]}
+            columns={[
+              { field: 'brand', valueGetter: (params) => params.value.year, type: 'number' },
+            ]}
+          />,
+        );
+        expect(getColumnValues()).to.deep.equal(expected.map((res) => res.toLocaleString()));
+      });
+    });
+  });
+
+  describe('Date operators', function test() {
+    const isEdge = /Edg/.test(window.navigator.userAgent);
+    before(function before() {
+      if (isEdge) {
+        // We need to skip edge as it does not handle the date the same way as other browsers.
+        this.skip();
```

Comment:
This problem is not with Edge but with BrowserStack. All Edge versions on it are with locale set to `en_GB` instead of `en_US`. Unfortunately, the karma launcher doesn't have an option to change the locale. Related to https://github.com/karma-runner/karma-browserstack-launcher/issues/187.

Reply:
OK, so it sounds very much like a bug with the filtering logic. The test should be timezone invariant, we don't control the timezone of the end users.